<launch>
    <!-- Launch the RPLidar node -->
    <node name="rplidarNode" pkg="rplidar_ros" type="rplidarNode" output="screen">
        <param name="serial_port" type="string" value="/dev/ttyUSB1"/>
        <param name="serial_baudrate" type="int" value="115200"/>
        <param name="frame_id" type="string" value="laser"/>
        <param name="inverted" type="bool" value="false"/>
        <param name="angle_compensate" type="bool" value="true"/>
    </node>

    <!-- Debug parameters -->
    <arg name="debug" default="false" doc="Enable debug mode with test map"/>
    <arg name="debug_map_type" default="empty" doc="Type of debug map: empty, obstacles, maze"/>
    <arg name="use_slam" default="true" doc="Enable SLAM and pose tracking"/>

    <!-- Start the obstacle avoidance node -->
    <node name="obstacle_avoidance_node" pkg="robot_navigation" type="client.py" output="screen">
        <!-- Pass parameters to the node -->
        <param name="debug" value="$(arg debug)"/>
        <param name="debug_map_type" value="$(arg debug_map_type)"/>
        <param name="safety_distance" value="0.1"/>
        <param name="smoothing_weight" value="0.5"/>
        <param name="turning_cost" value="0.3"/>
        <param name="use_slam" value="$(arg use_slam)"/>
    </node>

    <!-- Start RViz with our configuration -->
    <node name="rviz" pkg="rviz" type="rviz" 
          args="-d $(find robot_navigation)/rviz/obstacle_avoidance.rviz"/>

    <!-- Static transforms -->
    <node pkg="tf2_ros" type="static_transform_publisher" name="map_to_odom"
          args="0 0 0 0 0 0 map odom" />
          
    <node pkg="tf2_ros" type="static_transform_publisher" name="base_to_laser"
          args="0 0 0 0 0 0 base_link laser_frame" />

</launch>